lib/pos_blockchain.go

```diff
@@ -172,25 +172,14 @@ func (bc *Blockchain) processOrphanBlockPoS(block *MsgDeSoBlock) error {
 		if err != nil {
 			return errors.Wrapf(err, "processOrphanBlockPoS: Problem getting snapshot at epoch number for poch entry at epoch #%d", epochEntry.EpochNumber)
 		}
-		// Okay now that we've gotten the SnapshotAtEpochNumber for the next epoch, we can make sure that the proposer
-		// of the block is within the set of potential block proposers for the next epoch.
-		snapshotLeaderPKIDs, err := utxoView.GetSnapshotLeaderScheduleAtEpochNumber(epochEntrySnapshotAtEpochNumber)
+		// Okay now that we've gotten the SnapshotAtEpochNumber for the prev/next epoch, we can make sure that the proposer
+		// of the block is within the set of potential block proposers for the prev/next epoch based on
+		// the VotingPublicKey
+		snapshotValidatorEntry, err := utxoView.GetSnapshotValidatorEntryByBLSPublicKey(block.Header.ProposerVotingPublicKey, epochEntrySnapshotAtEpochNumber)
 		if err != nil {
-			return errors.Wrapf(err, "processOrphanBlockPoS: Problem getting snapshot leader schedule at snapshot at epoch number %d", epochEntrySnapshotAtEpochNumber)
+			return errors.Wrapf(err, "processOrphanBlockPoS: Problem getting snapshot validator entry for block proposer %v", block.Header.ProposerVotingPublicKey)
 		}
-		// Get the PKID for the block proposer.
-		blockProposerPKID := utxoView.GetPKIDForPublicKey(block.Header.ProposerPublicKey.ToBytes()).PKID
-		// TODO: Replace w/ collections.Any for simplicity. There is an issue with this version
-		// of Go's compiler that is preventing us from using collections.Any here.
-		// We can now check if the block proposer is in the set of snapshot leader PKIDs.
-		blockProposerSeen := false
-		for _, snapshotLeaderPKID := range snapshotLeaderPKIDs {
-			if snapshotLeaderPKID.Eq(blockProposerPKID) {
-				blockProposerSeen = true
-				break
-			}
-		}
```

Comment:
Why remove the check for the block proposer's existence in the leader schedule? Assuming you perform the check elsewhere, but I just want to make sure